## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 513111`
Commit: [`ffb41094fa908de418bbcd0482a85060d7e862a9`](https://github.com/NixOS/nixpkgs/commit/ffb41094fa908de418bbcd0482a85060d7e862a9) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/ffb41094fa908de418bbcd0482a85060d7e862a9..pull/513111/head))
Merge: [`730d8f2ed1566e915b30a905f0b85f4c6d54be56`](https://github.com/NixOS/nixpkgs/commit/730d8f2ed1566e915b30a905f0b85f4c6d54be56)

Logs: https://github.com/ToasterUwU/nixpkgs-review-gha/actions/runs/24899457080


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>xr-chaperone</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>xr-chaperone</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>xr-chaperone</summary>
<pre>
error[E0308]: mismatched types
   --&gt; src/xr_session.rs:68:17
    |
 66 |             xr_instance.create_vulkan_device(
    |                         -------------------- arguments to this method are incorrect
 67 |                 system_id,
 68 |                 instance_addr,
    |                 ^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected fn pointer `unsafe extern &quot;system&quot; fn(_, *const u8) -&gt; std::option::Option&lt;_&gt;`
               found fn pointer `unsafe extern &quot;system&quot; fn(_, *const i8) -&gt; std::option::Option&lt;_&gt;`
note: method defined here
   --&gt; /build/xr-chaperone-0-unstable-2026-03-16-vendor/source-registry-0/openxr-0.21.1/src/instance.rs:344:19
    |
344 |     pub unsafe fn create_vulkan_device(
    |                   ^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `xr-chaperone` (bin &quot;xr-chaperone&quot;) due to 2 previous errors</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>xr-chaperone</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>xr-chaperone</summary>
<pre>103 |     use libc::Elf64_Addr as ElfAddr;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Elf64_Addr` in the root

   Compiling tracing-subscriber v0.3.22
error[E0425]: cannot find function `dlinfo` in crate `libc`
   --&gt; /nix/build/nix-12213-3422198853/xr-chaperone-0-unstable-2026-03-16-vendor/source-registry-0/libmonado-1.6.0/src/lib.rs:116:9
    |
116 |         libc::dlinfo(
    |               ^^^^^^ not found in `libc`

error[E0425]: cannot find value `RTLD_DI_LINKMAP` in crate `libc`
   --&gt; /nix/build/nix-12213-3422198853/xr-chaperone-0-unstable-2026-03-16-vendor/source-registry-0/libmonado-1.6.0/src/lib.rs:118:10
    |
118 |             libc::RTLD_DI_LINKMAP,
    |                   ^^^^^^^^^^^^^^^ not found in `libc`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `libmonado` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>xr-chaperone</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>xr-chaperone</summary>
<pre>    |
103 |     use libc::Elf64_Addr as ElfAddr;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Elf64_Addr` in the root

error[E0425]: cannot find function `dlinfo` in crate `libc`
   --&gt; /nix/build/nix-8098-3596483442/xr-chaperone-0-unstable-2026-03-16-vendor/source-registry-0/libmonado-1.6.0/src/lib.rs:116:9
    |
116 |         libc::dlinfo(
    |               ^^^^^^ not found in `libc`

error[E0425]: cannot find value `RTLD_DI_LINKMAP` in crate `libc`
   --&gt; /nix/build/nix-8098-3596483442/xr-chaperone-0-unstable-2026-03-16-vendor/source-registry-0/libmonado-1.6.0/src/lib.rs:118:10
    |
118 |             libc::RTLD_DI_LINKMAP,
    |                   ^^^^^^^^^^^^^^^ not found in `libc`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `libmonado` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/xr/xr-chaperone/package.nix
+++ b/pkgs/by-name/xr/xr-chaperone/package.nix
@@ -1,0 +1,54 @@
+{
+  rustPlatform,
+  fetchFromGitHub,
+  autoPatchelfHook,
+  cmake,
+  git,
+  python3,
+  openxr-loader,
+  libxkbcommon,
+  lib,
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "xr-chaperone";
+  version = "0-unstable-2026-03-16";
+
+  src = fetchFromGitHub {
+    owner = "FrostyCoolSlug";
+    repo = "xr-chaperone";
+    rev = "bb3a78dc8abc802f51b4a257a019b31e0e03c940";
+    hash = "sha256-djMOvA1XIYlQgXxBNEbCbSdHAMOmyUGTVqeIEV8Nv3c=";
+  };
+
+  cargoHash = "sha256-9uEosKwKGNruwxp/uslXj0WAFowY4Tu2CikWa2JiOf4=";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cmake
+    git
+    python3
+  ];
+
+  buildInputs = [ openxr-loader ];
+
+  runtimeDependencies = [
+    libxkbcommon
+  ];
+
+  postInstall = ''
+    install -m 444 -D resources/xr-chaperone.desktop $out/share/applications/${finalAttrs.pname}.desktop
+    install -m 444 -D resources/xr-chaperone.png $out/share/icons/hicolor/256x256/apps/${finalAttrs.pname}.png
+    install -m 444 -D resources/xr-chaperone.svg $out/share/icons/hicolor/scalable/apps/${finalAttrs.pname}.svg
+  '';
+
+  __structuredAttrs = true;
+
+  meta = {
+    description = "VR Chaperone System for OpenXR ";
+    homepage = "https://github.com/FrostyCoolSlug/xr-chaperone";
+    license = lib.licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "xr-chaperone";
+    maintainers = with lib.maintainers; [ toasteruwu ];
+  };
+})

